gm4_lively_lily_pads/data/gm4_lively_lily_pads/function/mechanics/interactions/placement/interact_rcd.mcfunction

```diff
@@ -0,0 +1,44 @@
+# place interaction logic
+# @s = player who right clicked on interaction holding a supported item
+# at @s
+# run from advancement: placement_rcd
+
+advancement revoke @s only gm4_lively_lily_pads:placement_rcd
+
+# breif check to enable Upgrade_Path 3.0
+execute if entity @e[type=minecraft:interaction,tag=lilyPadInt,limit=1] run scoreboard players set lively_lily_pads gm4_earliest_version 0
```

Comment:
This has a pretty big performance impact due to the lack of a distance argument. I'd strongly recommend a max distance of 64 blocks.